docs/source/guides/data.rst

```diff
@@ -252,9 +252,61 @@ set is not available at the time your model was declared. Simply pass the test s
 
 --------------
 
+********************************************
+Accessing DataLoaders within LightningModule
+********************************************
+
+In the case that you require access to the DataLoader or Dataset objects, DataLoaders for each step can be accessed using the ``Trainer`` object:
+
+.. code-block:: python
+
+    def test_step(self, batch, batch_idx, dataloader_idx):
+        test_dl = self.trainer.test_dataloaders[dataloader_idx]
+        test_dataset = test_dl.dataset
+        # extract metadata, etc. from the dataset:
+        ...
+
+Since Lightning automatically takes care of iterating over the DataLoader, this is generally unnecessary unless you need to access metadata.
```

Comment:
```suggestion
Since PyTorch Lightning automatically takes care of iterating over the DataLoader, this is generally unnecessary unless you need to access metadata.
```

Reply:
I think just Lightning tells it's PyTorch Lightning. That's what we use at other places as well.